internal/service/advancedcluster/resource_advanced_cluster.go

```diff
@@ -578,7 +578,11 @@ func resourceRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Di
 
 	zoneNameToOldReplicationSpecMeta, err := GetReplicationSpecAttributesFromOldAPI(ctx, projectID, clusterName, connV220240530.ClustersApi)
 	if err != nil {
-		return diag.FromErr(err)
+		if apiError, ok := admin20240530.AsError(err); !ok {
+			return diag.FromErr(err)
+		} else if !(apiError.GetErrorCode() == "ASYMMETRIC_SHARD_UNSUPPORTED" && isUsingOldShardingConfiguration(d)) {
+			return diag.FromErr(fmt.Errorf(errorRead, clusterName, err))
+		}
```

Comment:
For the case of `apiError.GetErrorCode() == "ASYMMETRIC_SHARD_UNSUPPORTED" && isUsingOldShardingConfiguration(d)`, I believe we should throw an error as this is the existing behaviour (mentioned in [tech design](https://docs.google.com/document/d/1wEnxQW89-Efb0i8JM7q6-GpXS_GSR4c325cnlN_YpIE/edit?tab=t.0#bookmark=id.b8588bhps0lg)).
I think the final criteria for returning an error would be apiError != "ASYMMETRIC_SHARD_UNSUPPORTED" || (apiError == "ASYMMETRIC_SHARD_UNSUPPORTED" && isUsingOldShardingConfiguration(d)).
Same criteria can be applied for singular data source.